File: sharedModule/src/main/java/ru/nekit/android/qls/pupil/avatar/IPupilAvatarPart.java
package ru.nekit.android.qls.pupil.avatar;

import android.support.annotation.DimenRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public interface IPupilAvatarPart {

    int getVariant(int position);

    int getVariantCount();

    @DimenRes
    int getY();

    @NonNull
    String getPartName();

    @Nullable
    IPupilAvatarPart[] getDependentItems();

    @NonNull
    PupilAvatarPartType getPupilAvatarPartType();

}

File: sharedModule/src/main/java/ru/nekit/android/qls/pupil/avatar/IPupilAvatarPart.kt
package ru.nekit.android.qls.pupil.avatar

import android.support.annotation.DimenRes
import android.support.annotation.DrawableRes

interface IPupilAvatarPart {

    @get:DimenRes
    val y: Int

    val name: String

    val dependentItems: Array<IPupilAvatarPart>?
    val pupilAvatarPartType: PupilAvatarPartType

    @get:DrawableRes
    val variants: IntArray

}
